source/core/ut_output_buffer.pkb

```diff
@@ -0,0 +1,100 @@
+create or replace package body ut_output_buffer is
+
+  procedure send_line(a_reporter ut_reporter_base, a_text varchar2) is
+    l_text_list  ut_varchar2_list;
+    pragma autonomous_transaction;
+  begin
+    if a_reporter is not null and a_reporter.reporter_id is not null and a_reporter.start_date is not null and a_text is not null then
+      if length(a_text) > ut_utils.gc_max_storage_varchar2_len then
+        l_text_list := ut_utils.clob_to_table(a_text, ut_utils.gc_max_storage_varchar2_len);
+        forall i in 1 .. l_text_list.count
+          insert into ut_output_buffer_tmp(start_date, reporter_id, message_id, text)
+          values (a_reporter.start_date, a_reporter.reporter_id, ut_message_id_seq.nextval, l_text_list(i));
+      else
+        insert into ut_output_buffer_tmp(start_date, reporter_id, message_id, text)
+        values (a_reporter.start_date, a_reporter.reporter_id, ut_message_id_seq.nextval, a_text);
+      end if;
+      commit;
+    end if;
+  end;
+
+  procedure close(a_reporter ut_reporter_base) is
+    pragma autonomous_transaction;
+  begin
+    insert into ut_output_buffer_tmp(start_date, reporter_id, message_id, is_finished)
+    values (a_reporter.start_date, a_reporter.reporter_id, ut_message_id_seq.nextval, 1);
+    commit;
+  end;
+
+  function get_lines(a_reporter_id varchar2, a_timeout_sec naturaln := gc_max_wait_sec) return ut_varchar2_list pipelined is
+    pragma autonomous_transaction;
+    l_results        ut_varchar2_list;
+    l_wait_wait_time number(10,1) := 0;
+    c_sleep_time     number(1,1)  := 0.1; --sleep for 100 ms between checks
+    l_finished       boolean := false;
+  begin
+    loop
+      delete from (
+        select *
+          from ut_output_buffer_tmp where reporter_id = a_reporter_id order by message_id
+        )
+      returning text bulk collect into l_results;
+
+      --nothing fetched from output, wait and try again
+      if l_results.count = 0 then
+        dbms_lock.sleep(c_sleep_time);
```

Comment:
We need to seep for a short while to avoid bombarding database with delete statements when zero rows produced by the run.